### ISSUE TITLE
Incorrect definition of zoom factor in high_na_ff_lens

### ISSUE THREAD

**Issue by nikolaikarner** [chromatix-team/chromatix#185]
The zoom_factor factor defined in chromatix/functional/lenses : high_na_ff lens is incorrect since the function returns a focal-plane field whose reconstructed size depends on the input array size N and the physical simulation window, even when the physical optical setup (aperture, focal lengths, wavelengths, output_shape, output_dx) is held fixed. This makes the function's output non-convergent: refining the simulation grid changes the answer. 

Small script for testing:
```

import jax.numpy as jnp
import chromatix.functional as cf
import matplotlib.pyplot as plt

lam = 633e-9      # wavelength
f = 3e-3          # folcal length of objective
NA = 1.49         # Numerical Aperture
n = 1.515         # Refractive index
window = 6e-3

N=2058
dx = window / N

field_pup = cf.plane_wave((N, N), dx, lam, scalar=True, pupil = lambda field: cf.circular_pupil(field, 2e-3))

extent_pupil = jnp.array([ -field_pup.extent[0]/2, field_pup.extent[0]/2, -field_pup.extent[1]/2, field_pup.extent[1]/2])
plt.imshow(jnp.log(field_pup.intensity), extent=extent_pupil*1e3)
plt.xlabel("mm")
plt.ylabel("mm")
plt.colorbar()
plt.show()


field_target = cf.high_na_ff_lens(
        field_pup, f, n, NA,
        output_dx=100e-9,            # FIXED
        output_shape=(1024,1024)     # FIXED
    )
I = jnp.array(field_target.intensity)

extent = jnp.array([-field_target.extent[0]/2,field_target.extent[0]/2, -field_target.extent[1]/2 , field_target.extent[1]/2])
plt.imshow(I, extent=extent*1e6)
plt.xlabel("um")
plt.ylabel("um")
plt.show()

```

If this is run with different values of N then scaling of the output field changes

**Comment by diptodip** [comment]
Thanks for catching this! I'll get this fixed soon.

**Comment by diptodip** [comment]
This is fixed in the [dd/dev](https://github.com/chromatix-team/chromatix/tree/dd/dev) branch. It will come to main with a new release soon.

EDIT: Actually, it needs a little more work but should be fixed in a bit with some clarifications in the docs.

**Comment by nikolaikarner** [comment]
Thank you for working on it. 
Another little bug I found is  in the utils/czt.py.zoomed_fft 

I think line 113 and 115 should be 
    w = tuple(jnp.exp(-1j * (k_end - k_start) / n) for n in renorm)
and
    a = jnp.exp(1j * k_start)
The signs of the exponents are wrong

You can see this by comparing to the default jnp fft

```

from chromatix.utils.czt import zoomed_fft

import jax.numpy as jnp
x = jnp.array([1.0, 2.0, 3.0+1j, 4.0-2j])  # complex, so conjugate-symmetry can't hide the bug
print(jnp.fft.fft(x))
print(zoomed_fft(x, 0.0, 2*jnp.pi, (4,), axes=(-1,)))
```

**Comment by diptodip** [comment]
No, thank you for testing these out! This is also fixed in [dd/dev](https://github.com/chromatix-team/chromatix/tree/dd/dev). Note that in order to match `jnp.fft.fft` you will need to set `include_end=False` in `zoomed_fft` (there's a new test to show this).

**Comment by nikolaikarner** [comment]
Thank you for the info.
I am still not entirely sure that the high_na_tube_lens now works correctly. 
The zoom factor seems to be fixed but the output phase seems a little weird sometimes. I mean in a low NA case it should give you a similar result to the normal ff_lens is that correct?

And what exactly is the issue that the field has to be exactly of the size of the pupil in the high na lens, why does it not work if one uses a pupil mask ?

**Comment by diptodip** [comment]
There are a few differences between `ff_lens` and `high_na_tube_lens`. To make the results match up you need to choose your `output_dx` in `high_na_tube_lens` to match what an FFT in `ff_lens` does (`output_dx` should be `λ f / (n · N · Δx_in)`). The high NA function also applies an apodization correction that `ff_lens` doesn't apply, which you could apply to the input field yourself if you wanted. Check out the new `test_high_na_tube_lens_matches_ff_lens` in `test_lenses.py` in the `dd/dev` branch.

There was also a global phase factor that wasn't being applied, which is probably the biggest difference you were seeing at low NA where the apodization correction doesn't make a huge difference. That's fixed now, thanks to you pointing it out. Note that because `high_na_tube_lens` uses CZT (and needs some extra output phase corrections to remove a phase ramp at the output which the FFT case does not have), you'll see some numerical error that can make the phase look a bit strange as well (depending on how you're sampling and whether you're zooming in or not).

RE: the note about the pupil size in the docstring, it's really just that you need to make sure the extent of the field is at least as big as the BFP diameter of the objective and that your sampling is fine enough that all the phase factors that will be applied when calculating the CZT are not aliased. The same is true of `ff_lens` and the FFT phase factors, of course. Bad wording on my part.

What I was trying to get at is that `high_na_tube_lens` was really designed to be used in the specific scenario of taking the field at the BFP of an objective lens as its input. That's also what I was trying to convey by renaming the function. The correction that the function applies and the zooming don't make sense otherwise.

It's awesome to have you test this so thoroughly! Let me know if you find anything else that looks weird.

**Comment by nikolaikarner** [comment]
Hi,
Thank you for the explanation. At the moment I am using it for the simulation of an objective lens altough I am really interested how much effect this has on lower NA fields becuase it seems that some optimizations of mine work better with the verctorial high NA lens. But first I want to make shure it works correctly. I attached a small script (and a utils file  for the plotting) which simulates a simple setup I am working at the moment which uses an SLM to create bleaching patterns for cells. This pattern then is demagnified and imaged onto the cell. I can see a clear difference here when using a ff_lens and using a high_na lens as objective lens. If you could explain this to me becuase now apply the apodization factor as you do in the test buty still see a different result.

**Comment by nikolaikarner** [comment]
I really like the chromatix package, thank you for putting in the work.

**Comment by nikolaikarner** [comment]
[test_setup.zip](https://github.com/user-attachments/files/30979594/test_setup.zip)
